### ISSUE TITLE
Build error using opencv

### ISSUE THREAD

**Issue by mallidi** [Microsoft/CNTK#427]
Hi,

I am trying to build CNTK WITHOUT opencv. This is how I am running configuration step

~> ./configure --cuda=yes --with-cuda=/opt/NVIDIA/cuda-7.0 --with-gdk=/export/a07/harish/cntk/GDK/usr --with-cub=/export/a07/harish/cntk/CUB/cub-1.4.1 --1bitsgd=yes --with-cudnn=/export/a07/harish/cntk/cudnn-4.0/ --with-acml=/export/a07/harish/cntk/ACML/acml5.3.1/ifort64_mp

As you can see I am not using --with-opencv option. So I am assuming build process should be running successfully without opencv libraries

When I do make , I get the following error

~> make
=-----------------------------------------------------------=
creating .build/Source/Readers/ImageReader/Exports.o for Linux with build type release
mpic++ -c Source/Readers/ImageReader/Exports.cpp -o .build/Source/Readers/ImageReader/Exports.o -D_POSIX_SOURCE -D_XOPEN_SOURCE=600 -D__USE_XOPEN2K -std=c++11  -DUSE_CUDNN -DUSE_ACML -DNDEBUG -DNO_SYNC -DQUANTIZED_GRADIENT_AGGREGATION  -msse3 -std=c++0x -fopenmp -fpermissive -fPIC -Werror -fcheck-new -Wno-error=literal-suffix -g -O4 -ISource/Common/Include -ISource/Math -ISource/CNTK -ISource/ActionsLib -ISource/ComputationNetworkLib -ISource/SGDLib -ISource/SequenceTrainingLib -ISource/CNTK/BrainScript -ISource/Readers/ReaderLib -I/export/a07/harish/cntk/GDK/usr/include/nvidia/gdk -I/export/a07/harish/cntk/CUB/cub-1.4.1 -I/opt/NVIDIA/cuda-7.0/include -I/export/a07/harish/cntk/cudnn-4.0//cuda/include -I/export/a07/harish/cntk/ACML/acml5.3.1/ifort64_mp/include -I/usr/include -ISource/1BitSGD -MD -MP -MF .build/Source/Readers/ImageReader/Exports.d
In file included from Source/Readers/ImageReader/ImageTransformers.h:10:0,
                 from Source/Readers/ImageReader/ImageReader.h:9,
                 from Source/Readers/ImageReader/Exports.cpp:12:
/usr/include/opencv2/opencv.hpp:58:39: fatal error: opencv2/contrib/contrib.hpp: No such file or directory
 #include "opencv2/contrib/contrib.hpp"
                                       ^
compilation terminated.
make: **\* [.build/Source/Readers/ImageReader/Exports.o] Error 1

Could you let me know how to fix this issue? should I run configure step with different settings?

**Comment by voges316** [comment]
Hi, I think I ran into the same issue.

tl;dr - I edit the Config.make file and delete the opencv_path line before running 'make all'

Opencv has some issues regarding cntk makefile
I have an older version of opencv (2.4.13) on Centos 6.8 vm for use with another program. CNTK (1.5.1) fails to compile with the older opencv version, but it compiles successfully if I remove/delete opencv completely.
I am trying to compile CNTK “without” opencv support while leaving my old opencv installation intact, but it appears the configure script detects the presence of the opencv libraries (mine are in /usr/local ) automatically and tries to compile with them. I was able to work around this by editing the Config.make file and removing the line that has the opencv folder and them compiling with ‘make all’. 

Following the installation instructions
mkdir build/release –p
cd build/release
../../configure
*** edit the Config.make file and delete the opencv_path line **
make all

Then CNTK compiled fine. There should prob be a configure option for 'opencv[=(yes/no)]'
